Makefile

```diff
@@ -306,7 +306,7 @@ artifacts: kustomize yq helm ## Generate release artifacts.
 	$(HELM) package --version $(GIT_TAG) --app-version $(GIT_TAG) charts/kueue -d artifacts/
 	mv artifacts/kueue-$(GIT_TAG).tgz artifacts/kueue-chart-$(GIT_TAG).tgz
 	# Revert the image changes
-	$(YQ)  e  '.controllerManager.manager.image.repository = "$(IMAGE_REGISTRY)/$(IMAGE_NAME)" | del(.controllerManager.manager.image.tag) | .controllerManager.manager.image.pullPolicy = "Always"' -i charts/kueue/values.yaml
+	$(YQ)  e  '.controllerManager.manager.image.repository = "$(STAGING_IMAGE_REGISTRY)/kueue/$(IMAGE_NAME)" | del(.controllerManager.manager.image.tag) | .controllerManager.manager.image.pullPolicy = "Always"' -i charts/kueue/values.yaml
```

Comment:
Why do we need this change? IIUC, if we specify `STAGING_IMAGE_REGISTRY=foo/bar make artifacts` will apply the custom registry here, isn't it?

https://github.com/kubernetes-sigs/kueue/blob/292054c57c4139c400064e05e5dc8f439cdae4bd/Makefile#L37-L38

Reply:
> Why do we need this change? IIUC, if we specify `STAGING_IMAGE_REGISTRY=foo/bar make artifacts` will apply the custom registry here, isn't it?

Our release procedure step says 

https://github.com/kubernetes-sigs/kueue/blob/1c9517d9975edf354a92a9c66054403b4960ebdf/.github/ISSUE_TEMPLATE/NEW_RELEASE.md?plain=1#L39

and this does not get reverted. I'm not sure you can override ` STAGING_IMAGE_REGISTRY := us-central1-docker.pkg.dev/k8s-staging-images ` like that.

Reply:
Rebased, ptal

Reply:
actually this happens only for full releases, when calling make artifacts with Image registry parameter as for the release version.

Reply:
> basically this command make artifacts IMAGE_REGISTRY=registry.k8s.io/kueue GIT_TAG=$VERSION, recommended in [our release steps](https://github.com/kubernetes-sigs/kueue/blob/main/.github/ISSUE_TEMPLATE/NEW_RELEASE.md?plain=1#L39), leaves the git status dirty.

Yes, I also always the working branch with `git reset --hard`. 

Thank you for describing that. I might misunderstand the objective for this change. 

If this PR aims to automatically revert the image change during the artifact preparation step, I'm fine with this PR. If the answer is yes, feel free to unhold PR.

/lgtm
/approve